zproject/prod_settings_template.py

```diff
@@ -120,12 +120,43 @@
     # 'zproject.backends.GoogleAuthBackend',  # Google auth, setup below
     # 'zproject.backends.GitHubAuthBackend',  # GitHub auth, setup below
     # 'zproject.backends.GitLabAuthBackend',  # GitLab auth, setup below
+    # 'zproject.backends.AppleAuthBackend',  # Apple auth, setup below
     # 'zproject.backends.AzureADAuthBackend',  # Microsoft Azure Active Directory auth, setup below
     # 'zproject.backends.SAMLAuthBackend', # SAML, setup below
     # 'zproject.backends.ZulipLDAPAuthBackend',  # LDAP, setup below
     # 'zproject.backends.ZulipRemoteUserBackend',  # Local SSO, setup docs on readthedocs
 )  # type: Tuple[str, ...]
 
+########
+# Apple OAuth.
+#
+# To set up Sign In with Apple, you'll need to do the following:
+#
+# (1) Visit https://developer.apple.com/account/resources/,
+# Follow the instructions in
+# https://help.apple.com/developer-account/?lang=en#/dev77c875b7e
+# to create a Sign In Certificate.
+#
+# (2) Follow instructions in
+# https://help.apple.com/developer-account/?lang=en#/dev1c0e25352.
+# to Create a Services ID. In services ID, fill the redirect URI with
+# a value like `http://zulip.example.com/complete/apple-id` .
+#
+# (3) You'll now have a client_id(eg: com.application.your), aka "Service ID",
+# a Key ID and a Private key file. Use the client_id as SOCIAL_AUTH_APPLE_ID_CLIENT
+# and Key ID as SOCIAL_AUTH_APPLE_ID_KEY and if you have a Team ID that
+# can be set as SOCIAL_AUTH_APPLE_ID_TEAM here.
+#
+# (4) Put the Private key file in zulip server in
+# /etc/zulip/apple/zulip-private-key.key. Make sure to set proper permissions
+# with the following commands
+# chown -R zulip.zulip /etc/zulip/apple/
```

Comment:
Let's use a colon in the argument to `chown`, not a dot -- so write `zulip:zulip`, not `zulip.zulip`.

TIL that a dot seems to work here! But that's undocumented (in at least the manpage I get with `man chown`, for GNU `chown`), and I suspect nonstandard. The conventional thing is a colon.